### PR TITLE
Convert setup

### PR DESCRIPTION
**Pull request by CSVdB**

### CODE REVIEW

File: smos-convert/package.yaml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+name:                smos-convert
+version:             0.1.0.0
+github:              https://github.com/NorfairKing/smos
+license:             BSD3
```
Comment (NorfairKing) [inline smos-convert/package.yaml]: AllRightsReserved

File: smos-convert/src/Import.hs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+module Import
+    ( module X
+    ) where
+
+import Prelude as X hiding (head, init, last, tail)
+
+import GHC.Generics as X hiding (to)
```
Comment (NorfairKing) [inline smos-convert/src/Import.hs]: Only import `Generic` instead of only hiding `to`.

File: smos-convert/src/Smos/Convert/OptParse.hs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+parseConvertFileArgs = ConvertFileArgs <$> parseOrgFilePath
+
+parseConvertFile :: ParserInfo Command
+parseConvertFile = info parser modifier
+  where
+    modifier = fullDesc <> progDesc "Write smosfile corresponding to an orgfile"
+    parser = ConvertFile <$> parseConvertFileArgs
+
+parseCommand :: Parser Command
+parseCommand = hsubparser $ command "convert" parseConvertFile
```
Comment (NorfairKing) [inline smos-convert/src/Smos/Convert/OptParse.hs]: So the command will be `smos-convert convert FILE`? Why not just `smos-convert FILE`?

File: smos-convert/test/Spec.hs [hunk 4]
```diff
@@ -0,0 +1,2 @@
+main :: IO ()
+main = putStrLn "Test suite not yet implemented"
```
Comment (NorfairKing) [inline smos-convert/test/Spec.hs]: Just leave it out then.

File: smos-data/src/Smos/Data/Types.hs [hunk 5]
```diff
@@ -126,7 +126,8 @@ instance FromJSON Entry where
 
 instance ToJSON Entry where
     toJSON Entry {..} =
-        if and [ isNothing entryContents
+        if and
+               [ isNothing entryContents
```
Comment (NorfairKing) [inline smos-data/src/Smos/Data/Types.hs]: No cosmetic changes here please.

**Review by NorfairKing — CHANGES_REQUESTED** [review]
Almost exactly the same review as last time. Please first review the PR yourself before requesting review.

File: smos-convert/test/Spec.hs (hunk 4 above)
Comment (CSVdB) [reply smos-convert/test/Spec.hs]: I don't understand why.
I added this as a reminder that I need to add tests.
What do you want to replace it by, and more importantly, what is wrong with this?

File: smos/src/Smos/Draw.hs [hunk 6]
```diff
@@ -55,7 +55,8 @@ drawNoContent :: Widget n
 drawNoContent =
     B.vCenterLayer $
     B.vBox $
-    map B.hCenterLayer
+    map
```
Comment (NorfairKing) [inline smos/src/Smos/Draw.hs]: No cosmetic changes.

**Review by NorfairKing — CHANGES_REQUESTED** [review]

**Comment by CSVdB** [comment]
I only changed the one comment you made.
Everything is good now.

**Pull request closed without merging**